wikipron/extract/core.py

```diff
@@ -31,8 +31,9 @@ def yield_pron(
         if not m:
             continue
         pron = m.group(1)
-        # Removes parens around various segments.
-        pron = pron.replace("(", "").replace(")", "")
+        # Removes parens around various segments unless --keep-parens is used.
+        if not config.parens:
+            pron = pron.replace("(", "").replace(")", "")
```

Comment:
Agreed that a unit test would be needed. If the key logic update is made here inside `yield_pron`, it might be tricky to write a test against just this. As I saw Kyle's comment wrt possibly adding an option to delete parenthetical material in the future, what do you think about a more future-proof approach along these lines:

* Define a function in `wikipron/extract/core.py` here, something like `_handle_pron(pron: str, config: Config) -> str:`. This function would have the `if not config.parens: pron = pron.replace("(", "").replace(")", "")` logic. Then in `yield_pron`, there would be `pron = _handle_pron(pron, config)` instead.
* If we do want to do other things like dropping parenthetical content later, then `_handle_pron` would be ready to cater for any new logic needed.
* Since `_handle_pron` is a standalone function, it'd be easily testable -- maybe the tests could live in `tests/test_wikipron/test_extract.py`?

Reply:
I was having similar thoughts re: the difficulty of testing. This seems like a good solution so I can definitely add it. The only thing is the name—do you think `_handle_paren` might be better, or should we keep it more general in case we want to open it up to other functionality beyond handling parenthetical content?

Reply:
> do you think _handle_paren might be better, or should we keep it more general in case we want to open it up to other functionality beyond handling parenthetical content?

The more specific `_handle_parens` is a great idea. If we need other "handle X" functions in the future, we can always add and test them separately as needed.

Reply:
+1 to adding a function for this.